asserts/snap_asserts.go

```diff
@@ -316,3 +316,142 @@ func assembleSnapRevision(assert assertionBase) (Assertion, error) {
 		timestamp:     timestamp,
 	}, nil
 }
+
+// Validation holds a validation assertion, describing that a combination of
+// (snap-id, approved-snap-id, approved-revision) has been validated for
+// the series, meaning updating to that revision of approved-snap-id
+// has been approved by the owner of the gating snap.
+type Validation struct {
+	assertionBase
+	revoked   bool
+	timestamp time.Time
+}
+
+// Series returns the series for which the validation holds.
+func (validation *Validation) Series() string {
+	return validation.HeaderString("series")
+}
+
+// SnapID returns the ID of the gating snap.
+func (validation *Validation) SnapID() string {
+	return validation.HeaderString("snap-id")
+}
+
+// ApprovedSnapID returns the ID of the gated snap.
+func (validation *Validation) ApprovedSnapID() string {
+	return validation.HeaderString("approved-snap-id")
+}
+
+// ApprovedSnapRevision returns the revision of the gated snap.
+func (validation *Validation) ApprovedSnapRevision() string {
+	return validation.HeaderString("approved-snap-revision")
+}
+
+// Revoked returns true if the validation has been revoked.
+func (validation *Validation) Revoked() bool {
+	return validation.revoked
+}
+
+// Timestamp returns the time when the validation was issued.
+func (validation *Validation) Timestamp() time.Time {
+	return validation.timestamp
+}
+
+// Implement further consistency checks.
+func (validation *Validation) checkConsistency(db RODatabase, acck *AccountKey) error {
+	_, err := db.Find(AccountType, map[string]string{
+		"account-id": validation.AuthorityID(),
+	})
+	if err == ErrNotFound {
+		return fmt.Errorf("validation assertion for snap id %q does not have a matching account assertion for the developer %q", validation.SnapID(), validation.AuthorityID())
+	}
+	if err != nil {
+		return err
+	}
+	_, err = db.Find(SnapDeclarationType, map[string]string{
+		"series":  validation.Series(),
+		"snap-id": validation.SnapID(),
+	})
+	if err == ErrNotFound {
+		return fmt.Errorf("validation assertion for snap-id %q does not have a matching snap-declaration assertion", validation.SnapID())
+	}
+	if err != nil {
+		return err
+	}
+	_, err = db.Find(SnapDeclarationType, map[string]string{
+		"series":  validation.Series(),
+		"snap-id": validation.ApprovedSnapID(),
+	})
+	if err == ErrNotFound {
+		return fmt.Errorf("validation assertion for approved-snap-id %q does not have a matching snap-declaration assertion", validation.ApprovedSnapID())
+	}
+	if err != nil {
+		return err
+	}
+	_, err = db.Find(SnapDeclarationType, map[string]string{
+		"series":  validation.Series(),
+		"snap-id": validation.SnapID(),
+	})
+	if err == ErrNotFound {
```

Comment:
I added a TestMissingGatingDeclaration that's similar to the snap revision one. 

To test missing the gated declaration is harder because I would need to have something like prereqDevAccount but for snap declaration "snap-id-1" and assertstest doesn't seem to have the right helper. Should I add it or it exists elsewhere and I missed it?